rtems/RTEMSThread.java

```diff
@@ -15,6 +15,8 @@ public RTEMSThread(int priority, Thread.State state){
 		this();
 		this.currentPriority = this.realPriority = priority;
 		this.state = state;
+		//this.lockMutex = new ChainControl();
+		PriorityQueue<RTEMSThread> wait = null;
```

Comment:
This still isn't fixed, the queue of RTEMSThread goes out of scope after the constructor, you want to remove the class declaration here to assign to the class instance variable. This line should probably be:

    wait = new PriorityQueue<RTEMSThread>();